packages/istanbul-lib-instrument/src/instrumenter.js

```diff
@@ -80,7 +80,13 @@ class Instrumenter {
         const opts = this.opts;
         const ast = babylon.parse(code, {
             allowReturnOutsideFunction: opts.autoWrap,
-            sourceType: opts.esModules ? "module" : "script"
+            sourceType: opts.esModules ? "module" : "script",
+            plugins: [
+              'asyncGenerators',
+              'dynamicImport',
+              'flow',
+              'jsx'
+            ]
```

Comment:
The list is taken from [here](https://github.com/babel/babylon#plugins). You can find all the "experimental" features of babylon under their test folder [here](https://github.com/babel/babylon/tree/master/test/fixtures/experimental). Adding some test won't be a problem, expecially if we take inspiration from their material. However I am not familiar enough with the plugin to write them.